### PR TITLE
MCR-85 Refactor menu.py

### PR DESCRIPTION
**Pull request by jasonlwj**
Addresses #85 (unless even more refactoring can be done? 👀)

### CODE REVIEW

File: catalogue.py [hunk 1]
```diff
@@ -69,6 +69,101 @@ def reset(self):
...
+                  to explain to the masses was significant. Once the Explorers got a taste, they have not been able to
+                  stop. They look higher, run farther and dig deeper, at the expense of their energy levels.
+                  Unfortunately for the Explorers, the required rest during the journey makes them easy targets for
+                  Heathens.""", "+ Units have 50% more stamina", "- Units have 75% of usual health",
+                  VictoryType.GLUTTONY),
+    FactionDetail(Faction.INFIDELS, """Some say they were raised by Heathens, and some say that their DNA is actually
+                  closer to Heathen than human. Regardless of their biological makeup, if you approach someone on the
+                  street of any settlement and bring up the Infidels, you will be met with a look of disgust and the
+                  question 'you're not one of them, are you?'. Seen as sub-human, other empires engage in combat on
+                  sight with the Infidels, no matter the disguises they apply.""", "+ Not attacked by heathens",
```
Comment (ChrisNeedham24) [inline catalogue.py:157]: Just realised I missed something in your last refactor, should be easy enough to address in this one - `Not attacked by heathens` should be replaced by `Special affinity with heathens`.

File: catalogue.py [hunk 2]
```diff
@@ -219,6 +314,16 @@ def reset(self):
     Faction.NOCTURNE: pyxel.COLOR_NAVY
 }
 
+# A map of victory types to their respective colours.
+VICTORY_TYPE_COLOURS: typing.Dict[Faction, int] = {
```
Comment (ChrisNeedham24) [inline catalogue.py:318]: Type hint is wrong here - should be `typing.Dict[VictoryType, int]`

File: catalogue.py [hunk 3]
```diff
@@ -69,6 +69,101 @@ def reset(self):
         """
         self.names = deepcopy(SETL_NAMES)
 
+# The list of playable factions and their details.
```
Comment (ChrisNeedham24) [inline catalogue.py:72]: Needs another new line above this comment.

File: menu.py [hunk 4]
```diff
@@ -149,64 +144,12 @@ def draw(self):
...
-                    case 13:
-                        pyxel.text(35, 70, "+ Thrive during the night", pyxel.COLOR_GREEN)
-                        pyxel.text(35, 90, "- Units weakened during the day", pyxel.COLOR_RED)
-                        pyxel.text(35, 120, "ELIMINATION", pyxel.COLOR_RED)
-
+                # Draw the buff and debuff text for the currently selected faction.
+                faction_detail = FACTION_DETAILS[self.faction_idx]
+                pyxel.text(35, 70, faction_detail.buff, pyxel.COLOR_GREEN)
+                pyxel.text(35, 90, faction_detail.debuff, pyxel.COLOR_RED)
+                pyxel.text(25, 170, faction_detail.rec_victory_type,
```
Comment (ChrisNeedham24) [inline menu.py:151]: The x and y coords are a bit cooked here, should be `35, 120` rather than `25, 170`.

File: menu.py [hunk 5]
```diff
@@ -870,22 +618,32 @@ def previous_menu_option(self,
...
-            case MainMenuOption():
-                self.main_menu_option = target_option
-            case VictoryType():
-                self.victory_type = target_option
+            case SetupOption(): self.setup_option = target_option
+            case WikiOption(): self.wiki_option = target_option
+            case MainMenuOption(): self.main_menu_option = target_option
+            case VictoryType(): self.victory_type = target_option
+
+    def get_option_colour(self, option: MenuOptions) -> pyxel.Colors:
```
Comment (ChrisNeedham24) [inline menu.py:633]: Couple of things with this one: if we're going to use that as the return type, I believe it has to be `pyxel.colors`. Also, semantically I'm not sure if that's what we really want, given that `pyxel.colors` is actually a list of integers, and we're returning just an integer here, not a list.

File: menu.py [hunk 6]
```diff
@@ -870,22 +618,32 @@ def previous_menu_option(self,
...
+        :param option: the menu option to pick the colour for.
+        :return: The appropriate colour.
+        """
+        match option:
+            case SetupOption(): field_to_check = self.setup_option
+            case WikiOption(): field_to_check = self.wiki_option
+            case MainMenuOption(): field_to_check = self.main_menu_option
+            case VictoryType(): field_to_check = self.victory_type
+
+        return pyxel.COLOR_RED if field_to_check is option else pyxel.COLOR_WHITE
```
Comment (ChrisNeedham24) [inline menu.py:646]: Pycharm is complaining that `field_to_check` might be undefined, which is true. Maybe we could just make the last case the default one?

File: menu.py [hunk 7]
```diff
@@ -44,6 +46,7 @@ class WikiOption(Enum):
     UNITS = "UNITS"
     BACK = "BACK"
 
+MenuOptions = SetupOption | WikiOption | MainMenuOption | VictoryType
```
Comment (ChrisNeedham24) [inline menu.py:49]: Need an extra new line both above and below this.

**Review by ChrisNeedham24 — CHANGES_REQUESTED** [review]
Broadly, changes look pretty good, just need to address a few things.

File: menu.py (hunk 6 above)
Comment (jasonlwj) [reply menu.py:646]: Won't be a problem if you change IDEs 🧠

**Review by ChrisNeedham24 — APPROVED** [review]
LGTM! Extra credit for voluntarily refactoring something.

**Pull request merged**